iOS/Collection View Management /CollectionViewDataSourceCreator.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  CollectionViewDataSource.swift
+//  CollectionViewDataSource
+//
+//  Created by Martin Young on 8/23/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import UIKit
+
+/// A protocol for types that can create a UICollectionViewDiffableDataSource.
+/// Implementing types provide a means to dequeue collection view cells and supplementary views.
+/// A data source creation function is automatically implemented by default and uses the dequeing functions
+/// to initialize a new datasource.
+protocol CollectionViewDataSourceCreator {
```

Comment:
Implementing types are responsible for creating diffable datasources and defining CellProvider functions. Importantly, DataSourceCreators DON'T hold onto any state. They don't hold a reference to the DiffableDataSource. They only create it.

I realized that the DiffableDataSource holds all the state we need and it only confuses things to wrap it up in another object. Instead we can interact with the DiffableDataSource directly and add convenience extensions if we find it awkward to manipulate the data.